File: kronos-idea-plugin/src/main/java/com/kotlinorm/plugin/idea/MainWinFactory.java
/**
 * Copyright 2022-2026 kronos-orm
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.kotlinorm.plugin.idea;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowFactory;
import com.intellij.ui.content.Content;
import com.intellij.ui.content.ContentFactory;

public final class MainWinFactory implements ToolWindowFactory {
    @Override
    @SuppressWarnings("PMD.LawOfDemeter")
    public void createToolWindowContent(Project project, ToolWindow toolWindow) {
        MainPanel mainPanel = new MainPanel(project);
        Content content = ContentFactory.getInstance().createContent(mainPanel.getComponent(), "", false);
        toolWindow.getContentManager().addContent(content);
    }
}

File: kronos-idea-plugin/src/main/kotlin/com/kotlinorm/plugin/idea/MainWinFactory.kt
package com.kotlinorm.plugin.idea

import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.ui.content.ContentFactory

class MainWinFactory : com.intellij.openapi.wm.ToolWindowFactory {
    override fun createToolWindowContent(
        project: Project,
        toolWindow: ToolWindow
    ) {
        val mainPanel = MainPanel(project)
        val content = ContentFactory.getInstance().createContent(mainPanel.component, "", false)
        toolWindow.contentManager.addContent(content)
    }
}